crates/katana/executor/src/implementation/blockifier/output.rs

```diff
@@ -74,6 +75,10 @@ impl TransactionExecutionOutput for TransactionExecutionInfo {
     fn revert_error(&self) -> Option<&str> {
         self.inner.revert_error.as_deref()
     }
+
+    fn execution_info(&self) -> TxExecInfo {
```

Comment:
we should cache the `TxExecInfo` when the method is first called so that subsequent call to this method won't require doing another conversion

Reply:
we can simply add an optional field to the `TransactionExecutionInfo` struct for each of the receipt and exec info to cache them on the first call of their specific methods.

i was initially thinking like this:

```rust
#[derive(Debug, Default)]
pub struct TransactionExecutionInfo {
    pub gas_used: u128,
    pub(super) inner: objects::TransactionExecutionInfo,
    receipt: Option<Receipt>,
    info: Option<TxExecInfo>
}

impl TransactionExecutionInfo {
    pub fn receipt(&self) -> &Receipt {
        if let Some(ref receipt) = self.receipt {
            receipt
        } else {
            // the original code here
            let receipt = .... ;
            self.receipt.get_or_insert(receipt)
        }
    }
}
```

but just realized that doing this would require marking the method as mutable...

Reply:
maybe we can ignore this for now and address it in subsequent PRs. this should not block your progress.

Reply:
> we can simply add an optional field to the `TransactionExecutionInfo` struct for each of the receipt and exec info to cache them on the first call of their specific methods.
> 
> i was initially thinking like this:
> 
> ```rust
> #[derive(Debug, Default)]
> pub struct TransactionExecutionInfo {
>     pub gas_used: u128,
>     pub(super) inner: objects::TransactionExecutionInfo,
>     receipt: Option<Receipt>,
>     info: Option<TxExecInfo>
> }
> 
> impl TransactionExecutionInfo {
>     pub fn receipt(&self) -> &Receipt {
>         if let Some(ref receipt) = self.receipt {
>             receipt
>         } else {
>             // the original code here
>             let receipt = .... ;
>             self.receipt.get_or_insert(receipt)
>         }
>     }
> }
> ```
> 

Oh, got you, evaluating the option does the job, that's great. :)

> but just realized that doing this would require marking the method as mutable...

Hum, good point.

> maybe we can ignore this for now and address it in subsequent PRs. this should not block your progress.

If it's fine for you, let's move forward to unlock at least the trace API that I can connect with the execution trace module of Saya to then have a proof being generated.
But definitely there's room for improvements here. Appreciate your time and review with specific concerns on performances, as always. :pray:

Reply:
yes all good.. i will work on that on my side after we're able to merge this